view/src/components/common/EventParticpantCard/EventParticpantCard.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import Paper from '@mui/material/Paper';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TablePagination from '@mui/material/TablePagination';
+import TableRow from '@mui/material/TableRow';
+import Box from '@mui/material/Box'
+import { useState, useEffect } from 'react';
+
+interface User {
+  id: number
+  name: string
+  number: number
+  created_at: number
+  updated_at: number
+};
+
+interface Events {
+  id: number
+  name: string
+  description: string
+  max_attendee: number
+  start_at: number
+  end_at: number
+  created_at: number
+  updated_at: number
+  user: User[]
+};
+
+const EventParticpantCard = () => {
+  const [eventusers, setEventUsers] = useState<Events[]>()
+
+  useEffect(() => {
+    fetch(process.env.CSR_API_URI + '/events/7/users', {
+      method: 'GET',})
+    .then(res => res.json())
+    .then(data => {
+        setEventUsers(data)
+        console.log(data)
+    })
+    .catch(error => {
+      console.error('エラーが発生しました', error);
+    });
+  },[])
+
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(10);
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(+event.target.value);
+    setPage(0);
+  };
+
+  return (
+    <Box sx={{ maxHeight:444, maxWidth:363}}>
+    <Paper sx={{ overflow: 'hidden' }}>
+      <TableContainer sx={{ maxHeight: 444, maxWidth: 363}}>
+        <Table stickyHeader aria-label="sticky table" style={{tableLayout: "fixed"}}>
+          <TableHead>
+            <TableRow>
+              <TableCell>名前</TableCell>
+              <TableCell align="left">学籍番号</TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {eventusers?.user
+            .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+            .map(user => (
+            <TableRow
+              hover role="checkbox" tabIndex={-1}
+              key={user.id}
+              sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
+            >
+              <TableCell >
+                {user.name}
+              </TableCell>
+              <TableCell align="left">
+                {user.number}
+              </TableCell>
+            </TableRow>
+            ))}
+          </TableBody>
+        </Table>
+      </TableContainer>
+      <TablePagination
+        rowsPerPageOptions={[5,10,25,100,{ label: 'All', value: -1 }]}
+        component="div"
+        count={eventusers?.user.length}
```

Comment:
countはundefinedを許容しないのでエラーが出てますね
undefinedの時は0などの初期値を入れておくといいです！

```suggestion
        count={eventusers?.user.length || 0}
```

`?`はoptional Chainingといって、nullやundefinedな呼び出しを可能とする記号です
eventusersがundefinedだと結果全体はundefinedとなるので、numberしか許容しないcountでエラーが出るわけです
参考：https://qiita.com/ryo2132/items/12981f61db21112cb61c#---optional-chaining--

`eventusers?.user.length || 0`でundefinedの時なぜ0に確定するかというと、or論理式の短絡評価があるためです。
`const x = a || b` としたとき、orなので片方が真の時点でtrueになります。
なのでaがtrueの時はtrueで確定するためaを返し、aがfalseの時はbによって真偽値が確定するためbを返します。
この時true/falseを返すのではなく、a/bの値を返すため、今回`eventusers?.user.length || 0`とすることで、左辺がundefinedの時は0が返ってくるわけです。
参考：https://zenn.dev/fujii0112/articles/78fe69c4c30731#%E2%97%87-%E5%9F%BA%E6%9C%AC-1